codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -158,17 +158,20 @@ public void incrementWindowSize(ChannelHandlerContext ctx, Http2Stream stream, i
             // Update the stream window and write any pending frames for the stream.
             FlowState state = state(stream);
             state.incrementStreamWindow(delta);
-            frameSent = false;
+            needFlush = false;
             state.writeBytes(state.writableWindow());
-            if (frameSent) {
+            if (needFlush) {
                 flush();
             }
         }
     }
 
     @Override
-    public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream,
+    public ChannelFuture sendFlowControlledData(ChannelHandlerContext ctx, Http2Stream stream,
             ByteBuf data, int padding, boolean endStream, ChannelPromise promise) {
+        if (stream.state() == Http2Stream.State.IDLE) {
```

Comment:
If we must check state here then shouldn't the check ensure the state is `OPEN`?  Can you explain (possibly against...I think the previous comment was deleted) why the [DefaultHttp2ConnectionEncoder.wireData](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java#L164) check is not sufficient?